ui/app/components/clients/table.ts

```diff
@@ -102,6 +111,18 @@ export default class ClientsTable extends Component<Args> {
     this[action] = value;
   }
 
+  @action
+  async resetPagination() {
+    // setPageSize is intentionally NOT reset here so user changes to page size
+    // are preserved regardless of whether or not the table data updates.
+    this.renderPagination = false;
+    this.currentPage = 1;
+    //  WORKAROUND to manually re-render Hds::Pagination::Numbered to force update @currentPage
```

Comment:
Yes, this is weird. I've reached out to the HDS folks about it since I don't think the arg is updating for the pagination component correctly